src/app/g3w-ol/controls/interactioncontrol.js

```diff
@@ -227,36 +426,146 @@ proto._handleClick = function(evt) {
   }
 };
 
-proto.getIteraction = function() {
+proto.getInteraction = function() {
   return this._interaction;
 };
 
 /**
  * Method to set filter operation intersect or Contains
  */
-
-proto.setSpatialMethod = function(method='intersects'){
+proto.setSpatialMethod = function(method='intersects') {
   this.spatialMethod = method;
   this.dispatchEvent({
     type: 'change-spatial-method',
     spatialMethod: this.spatialMethod
   });
 };
 
-proto.getSpatialMethod = function(){
+proto.getSpatialMethod = function() {
   return this.spatialMethod;
 };
 
-proto.setLayers = function(layers=[]){
+proto.setLayers = function(layers=[]) {
   this.layers = layers;
 };
 
 /**
- * called when project change
- * @param layers
+ * @param { unknown | null } layer
+ *
+ * @since 3.8.0
+ */
+proto.setSelectedLayer = function(layer) {
+  ControlsRegistry.setSelectedLayer(layer);
+};
+
+/**
+ * @since 3.8.0
+ */
+proto.getSelectedLayer = function() {
+  return ControlsRegistry.getSelectedLayer();
+};
+
+/**
+ * @since 3.8.0
+ */
+proto.getExternalLayers = function() {
+  return ControlsRegistry.getExternalLayers();
+};
+
+/**
+ * @param { 'intersects' | 'within' } spatialMethod
+ * 
+ * @listens change-spatial-method
+ * 
+ * @since 3.8.0
+ */
+proto.handleChangeSpatialMethod = function(spatialMethod) {
+  let eventKey = null;
+
+  const unlistenSpatialMethodChange = () => {
+    ol.Observable.unByKey(eventKey);
+    eventKey = null;
+  };
+
+  this.on('toggled', ({toggled}) => {
+    if (true === toggled) {
+      eventKey = this.on('change-spatial-method', this.runSpatialQuery);
+    } else if (null !== eventKey) {
+      unlistenSpatialMethodChange();
+      // reset to default
+      this.setSpatialMethod(spatialMethod);
+      this.clear();
+    }
+  })
+};
+
+/**
+ * @since 3.8.0
+ */
+proto.watchLayer = function(expOrFn, callback) {
+  return VM.$watch(expOrFn, callback)
+};
+
+/**
+ * @returns {boolean}
+ *
+ * @since 3.8.0
+ */
+proto.isSelectedLayerVisible = function() {
+  return (
+    'function' === typeof this.getSelectedLayer().isVisible
+      ? this.getSelectedLayer().isVisible()                 // in case of a project project
+      : this.getSelectedLayer().visible                     // in case of external layer
+  )
+};
+
+/**
+ * @returns {boolean} whether at least one of stored `this.layers` is visible
+ *
+ * @since 3.8.0
+ */
+proto._hasVisibleLayer = function() {
+  return !!((this.layers.length > 0) && this.layers.find(layer => layer.isVisible()));
+};
+
+/**
+ * @returns {boolean} whether at least one of stored `this.getExternalLayers()` is visible
+ *
+ * @since 3.8.0
+ */
+proto._hasVisibleExternalLayer = function() {
+  return !!(this.getExternalLayers().find(layer => layer !== this.layer && true === layer.visible));
+};
+
+/**
+ * @since 3.8.0
+ */
+proto.isThereVisibleLayers = function() {
+  return !!(this._hasVisibleLayer() || this._hasVisibleExternalLayer());
+};
```

Comment:
- missing `@returns` type for `isThereVisibleLayers()` function
- consider renaming `isThereVisibleLayers()` into `hasVisibleLayers()` for greater consistency with other functions
- find out if we also need to rename some other functions as well (in order to have a easier to undestard conditions and avoid oversights) \*


\* right now it would be something like this: `hasVisibleLayers()` = `_hasVisibleLayer()` + `_hasVisibleExternalLayer()` (so it is quite difficult to understand the difference between private `_hasVisibleLayer()` and the outer public `hasVisibleLayers()` function)\*